test/extended/testdata/build-secrets/s2i-binary-dir/.s2i/bin/run

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Ensure none of the build config inject secrets still exist in the file system
+for s in $(ls /tmp/secret? secret?); do
```

Comment:
With the bash syntax you suggest, I believe the loop will execute twice even if no files match the patterns. 
For example:
for secret in /tmp/filenotpresent?  filenotpresent2?; do
       echo "Matched: $secret"
done

will output:
Matched: /tmp/filenotpresent?
Matched: filenotepresent2?

That is why I wrap in an ls normally. The behavior isn't destructive, in this case, but it's counter-intuitive to me.  
Would you still prefer the unwrapped version?